dbms/src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -569,6 +569,17 @@ class MergeTreeData : public ITableDeclaration
     MergeTreeData::MutableDataPartPtr cloneAndLoadDataPart(const MergeTreeData::DataPartPtr & src_part, const String & tmp_part_prefix,
                                                            const MergeTreePartInfo & dst_part_info);
 
+    DiskSpaceMonitor::ReservationPtr reserveSpaceAtDisk(UInt64 expected_size) const; ///@TODO_IGR ASK Maybe set this method as private?
+
+    Strings getFullPaths() const {
+        auto paths = schema.getFullPaths();
+        for (auto && path : paths) {
+            path += table_name + '/'; ///@TODO_IGR ASK It is too slow(
```

Comment:
Why do we need it?

Reply:
To iterate over all paths to
1) Remove all perts
2) Load all parts from disk